src/Plutarch/Extra/IsData.hs

```diff
@@ -108,6 +106,65 @@ newtype ProductIsData (a :: Type) = ProductIsData {unProductIsData :: a}
 -- | Variant of 'PConstantViaData' using the List repr from 'ProductIsData'
 newtype DerivePConstantViaDataList (h :: Type) (p :: S -> Type) = DerivePConstantViaDataList h
 
+type family GetPRecord' (a :: [[S -> Type]]) :: [PLabeledType] where
+    GetPRecord' '[ '[PDataRecord a]] = a
+
+type family GetPRecord (a :: S -> Type) :: S -> Type where
+    GetPRecord a = PDataRecord (GetPRecord' (PCode a))
+
+type family GetRecordTypes (n :: [[Type]]) :: [S -> Type] where
+    GetRecordTypes '[x ': xs ] = PConstanted x ': GetRecordTypes '[xs]
+    GetRecordTypes '[ '[]] = '[]
+
+type family PUnlabel (n :: [PLabeledType]) :: [S -> Type] where
+    PUnlabel ((_ ':= p) ': xs) = p ': PUnlabel xs
+    PUnlabel '[] = '[]
+
+type family MatchTypes' (n :: [S -> Type]) (m :: [S -> Type]) :: Bool where
+    MatchTypes' '[] '[] = 'True
+    MatchTypes' (x ': xs) (x ': ys) = MatchTypes' xs ys
+    MatchTypes' (x ': xs) (y ': ys) = 'False
+    MatchTypes' '[] ys = 'False
+    MatchTypes' xs '[] = 'False
+
+type family MatchTypesError (n :: [[Type]]) (m :: [S -> Type]) (a :: Bool) :: Constraint where
+    MatchTypesError _ _ 'True = ()
+    MatchTypesError n m 'False =
+        ( 'True ~ 'False
+        , TypeError
+            ( 'Text "More/Less types are given to the Plutarch type"
+                ':$$: 'ShowType (GetRecordTypes n)
```

Comment:
Could we get `Constituent Haskell Types:` and `Constituent Plutarch Types:` annotation here? It won't be apparent unless you've familiar with the types.

Also, can we make the `"More/Less types are given to the Plutarch type"` more descriptive? Maybe something like `"Error in 'PlutusTypeDataList': Mismatch between constituent Haskell and Plutarch types"`

Reply:
It'd be nice. Feel free to commit it yourself, because I'll have access to my desktop in few hours.

Reply:
let me figure out how XD

Reply:
it's just like string : D

Reply:
I did it.